samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -65,15 +70,19 @@
    * Key is chosen to be processorId since at a time only one placement action can be in progress on a container.
    */
   private final ConcurrentHashMap<String, ContainerPlacementMetadata> actions;
+  private final DequeuedPlacementActionsCache dequeuedActionsCache;
```

Comment:
This class seems un-neccessary can it be replaced with 
`private final Set<String> placementActionsInFlight`

Reply:
Refactored it to make generic BoundedFifoQueue<T>